src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -267,11 +285,71 @@ public override async ValueTask OnReferenceFoundAsync(SourceReferenceItem refere
                     var classifiedSpans = classifiedSpansAndHighlightSpan.ClassifiedSpans;
                     var docText = await documentSpan.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
 
-                    return new ClassifiedTextElement(
-                        classifiedSpans.Select(cspan => new ClassifiedTextRun(cspan.ClassificationType, docText.ToString(cspan.TextSpan))));
+                    var classifiedTextRuns = GetReferenceClassifiedTextRunsWithHighlight(originalSymbolDisplayName, classifiedSpans, docText, isWriteOperation);
+                    return new ClassifiedTextElement(classifiedTextRuns);
                 }
 
                 return null;
+
+                // Local functions
+                static ClassifiedTextRun[] GetDefinitionClassifiedTextRunsWithHighlight(string originalSymbolDisplayName, ClassifiedTextElement definitionText)
+                {
+                    using var _ = ArrayBuilder<ClassifiedTextRun>.GetInstance(out var classifiedTextRuns);
+                    foreach (var run in definitionText.Runs)
+                    {
+                        if (run.Text == originalSymbolDisplayName)
```

Comment:
er... let's not do this.  this means if we have a line with multiple values of that name, it will highlight multiple times.  

FAR should already give back vlaues that tell you what the span of the name is in that line.  We should be able to produce the runs from that, without having to back-infer it like this.

This approach will also not work for things like escaped named and whatnot.